src/lib/ext-popup.js

```diff
@@ -0,0 +1,111 @@
+import {
+  areExtsBeingMonitored,
+  startMonitorAll,
+  stopMonitorAll,
+  viewExtPage,
+} from '../lib/ext-listen.js';
+
+export default class Popup {
+  constructor() {
+    this.startMonitorAllBtn = document.getElementById('startMonitorBtn');
+    this.stopMonitorAllBtn = document.getElementById('stopMonitorBtn');
+    this.viewActivityLogBtn = document.getElementById('actLogPage');
+    this.monitorStatusText = document.getElementById('monitorStatus');
+    this.errorMsgText = document.getElementById('errorText');
+  }
+
+  renderErrorMsg(message) {
+    this.errorMsgText.textContent = message;
+    this.errorMsgText.style.display = 'block';
+  }
+
+  renderMonitorStartedUI() {
+    this.startMonitorAllBtn.setAttribute('disabled', 'disabled');
+    this.startMonitorAllBtn.classList.add('disabled');
+    this.stopMonitorAllBtn.removeAttribute('disabled');
+    this.stopMonitorAllBtn.classList.remove('disabled');
+
+    this.monitorStatusText.textContent = 'Extensions are being monitored';
+    this.monitorStatusText.classList.add('green');
+    this.monitorStatusText.classList.remove('red');
```

Comment:
Here, it would be better to use class names that reflects the meaning of the state the element should represent instead of the particular style being currently used, in other words: `"green"` could be `"success"` and `"red"` could be `"failure"` (or something like that).

This way if we change ideas about which style to apply, we don't need to change the class names being assigned to the elements.

Reply:
Yeah, thanks.